### PR TITLE
Added support for ECS Secrets

### PR DESCRIPTION
**Pull request by PettitWesley**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by PettitWesley** [comment]
@SoManyHs @allisaurus 

Regarding the mock changes, I ran `make generate` using the latest version of mockgen. When I run `make generate` multiple times on the code, there is no additional diff, so these changes seems to be stable. I suspect that mockgen has rejiggered how they create mocks a little bit, and that's why there's some extra changes. There's nothing weird with the license or year anymore though.

**Comment by PettitWesley** [comment]
Testing:

ECS Params file:

```
version: 1
task_definition:
  task_execution_role: ecsExecutionRole
  services:
    redis:
      secrets:
        - value_from: /mysecrets/dbpassword
          name: DB_PASSWORD
        - value_from: /mysecrets/dbusername
          name: DB_USERNAME
```

```
$ ecs-cli compose create
INFO[0000] Using ECS task definition                     TaskDefinition="secrets:1"
```

Results in this task def (only key fields are shown):

```
{
  "containerDefinitions": [
    {
      "linuxParameters": {
        "capabilities": {
          "add": null,
          "drop": null
        },
      },
      "secrets": [
        {
          "valueFrom": "/mysecrets/dbpassword",
          "name": "DB_PASSWORD"
        },
        {
          "valueFrom": "/mysecrets/dbusername",
          "name": "DB_USERNAME"
        }
      ],
      "image": "redis:latest",
      "essential": true,
      "name": "redis"
    }
  ],
  "taskDefinitionArn": "arn:aws:ecs:us-west-2:144718711470:task-definition/secrets:1",
  "family": "secrets",
  "requiresAttributes": [
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.docker-remote-api.1.17"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "ecs.capability.secrets.ssm.environment-variables"
    }
  ],
  "requiresCompatibilities": [
    "EC2"
  ],
  "revision": 1,
  "status": "ACTIVE",
}
```

File: ecs-cli/modules/utils/compose/ecs_params_reader_test.go [hunk 1]
```diff
@@ -85,7 +85,12 @@ task_definition:
     wordpress:
       essential: true
       repository_credentials:
-        credentials_parameter: arn:aws:secretsmanager:1234567890:secret:test-RT4iv`
+        credentials_parameter: arn:aws:secretsmanager:1234567890:secret:test-RT4iv
+      secrets:
+        - value_from: /mysecrets/dbpassword
```
[the remark this answers is not in the document]
Comment (PettitWesley) [reply ecs-cli/modules/utils/compose/ecs_params_reader_test.go]: Oh yeah, will do!

**Comment by ryanpark91** [comment]
This doesn't work on fargate with following error.
> The specified platform does not satisfy the task definition’s required capabilities.

**Comment by ajchida** [comment]
@ryanpark91 
"Important
This feature is not yet supported for tasks using the Fargate launch type."

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html

This feature is not supported on fargate (yet)

**Pull request merged**